static/app/utils/customerDomainOrgRoute.tsx

```diff
@@ -0,0 +1,36 @@
+import {Component} from 'react';
+
+import ConfigStore from 'sentry/stores/configStore';
+
+export default function customerDomainOrgRoute<P>(
+  WrappedComponent: React.ComponentType<P>
+) {
+  class CustomerDomainOrgRoute extends Component<P> {
+    render() {
+      const lastActiveOrganization = ConfigStore.get('lastOrganization');
+      const {organizationUrl} = ConfigStore.get('links');
+      const params = (this.props as any).params as Record<string, string> | undefined;
+      if (params) {
+        if (!('orgId' in params)) {
+          if (lastActiveOrganization) {
+            const currentHostname = new URL(String(window.location)).hostname;
+            const orgUrlHostname = new URL('/', organizationUrl).hostname;
+            if (currentHostname === orgUrlHostname) {
```

Comment:
Should we always use the currentHostname value? If users are using org slugs that are not theirs they will get errors from the API.